libs/community/langchain_community/llms/huggingface_hub.py

```diff
@@ -108,21 +128,20 @@ def _call(
         """
         _model_kwargs = self.model_kwargs or {}
         params = {**_model_kwargs, **kwargs}
-        response = self.client(inputs=prompt, params=params)
+
+        response = self.client.post(
+            json={"inputs": prompt, "params": params}, task=self.task
+        )
+        response = json.loads(response.decode())
```

Comment:
Here I would use the task-specific methods instead of `client.post` directly. For the record, `client.post(...)` is what's used internally by the client to make the calls. It is perfectly valid to use it but you also have higher-level method -like `client.text_generation(...)`- that are more appropriate to handle inputs and outputs correctly. See supported tasks and their documentation [on this page](https://huggingface.co/docs/huggingface_hub/guides/inference#supported-tasks).

So instead of those two lines, I would suggest something like 

```py
if self.task == "text-generation":
    response = client.text_generation(...)
elif self.task == "conversational"
    response = client.conversational(...)
...
```
Inputs and outputs may vary depending on the selected task. It would also avoid the logic `if isinstance(response, list): ... else: ...` below that might be error prone.

Reply:
Hi @Wauplin, thanks for your precious feedback! I had a look at the `InferenceClient` implementation in hf-hub, but I couldn't find any method handling 'text2text-generation'. This task is still supported by `post` though. 
```py
        if self.task == "text-generation":
            response = self.client.text_generation(prompt, **params)
        elif self.task == "conversational":
            response = self.client.conversational(prompt, **params)["generated_text"]
        elif self.task == "translation":
            response = self.client.translation(prompt, **params)
        elif self.task == "summarization":
            response = self.client.summarization(prompt, parameters=params)
        else:
            raise ValueError(f"Invalid task {self.task}")
```

Reply:
Looks good to me yes! It should be more robust as well.
About `text2text-generation`, we are trying to deprecate it in favor of `text-generation` (for harmonization) cc @osanseviero. _I think_ that using `self.client.text_generation` should be fine but would be best to test it first. According to the docs, the API should be the same (see [here](https://huggingface.co/docs/api-inference/detailed_parameters#text2text-generation-task)).

Reply:
The API is the same, but the underlying pipelines in `transformers` are different (one is `encoder-decoder` while the other is `decoder`-only), so not sure they would work out of the box.

Reply:
@hwchase17 @baskaryan Any thoughts/suggestions on this?